File: api/app/domain/external/file_storage.py
from typing import Tuple
from fastapi import UploadFile
from typing import BinaryIO
from typing import Protocol
from app.domain.models.file import File


class FileStorage(Protocol):
    async def upload_file(self, uploadFile: UploadFile) -> File: ...

    async def download_file(self) -> Tuple[BinaryIO, File]: ...

File: api-node/src/infrastructure/file-storage/file-storage.ts
import { FileRecord } from '../../domain/models';

export interface FileStorage {
  uploadFile(file: Express.Multer.File): Promise<FileRecord>;
  getFileInfo(fileId: string): Promise<FileRecord>;
  downloadFile(fileId: string): Promise<{ buffer: Buffer; file: FileRecord }>;
}
